aten/src/ATen/TensorIterator.cpp

```diff
@@ -857,7 +857,7 @@ TensorIterator TensorIterator::reduce_op(Tensor& out, const Tensor& a) {
 TensorIterator TensorIterator::reduce_op(Tensor& out1, Tensor& out2, const Tensor& a) {
   TORCH_INTERNAL_ASSERT(out1.defined());
   TORCH_INTERNAL_ASSERT(out2.defined());
-  TORCH_CHECK((!a.is_cuda() && !out1.is_cuda() && !out2.is_cuda()) || (a.device() == out1.device() && out1.device() == out2.device()),
+  TORCH_CHECK((a.is_cpu() && out1.is_cpu() && out2.is_cpu()) || (a.device() == out1.device() && out1.device() == out2.device()),
```

Comment:
looks like this is not needed at all, because this will be handled in `build` since `check_all_same_device_ = true`